kernel/ipi.cc

```diff
@@ -90,3 +90,53 @@ on_ipicall()
     }
   }
 }
+
+std::atomic<int> paused_cpu_counter {0};
+enum pause_state { Ready, Pausing, Paused, Unpausing, Unpaused };
+std::atomic<enum pause_state> pause_state {Ready};
```

Comment:
It isn't great to give an enum and a variable the same name, since in C++ the "enum" (and "struct" / "class") keywords are optional when declaring a variable. Instead perhaps:

```cpp
enum pause_state_t { Ready, Pausing, Paused, Unpausing, Unpaused };
std::atomic<pause_state_t> pause_state {Ready}; 
```